src/hooks/use-user-info.jsx

```diff
@@ -0,0 +1,23 @@
+import { useCallback } from 'react'
+
+import { userService } from '~/services/user-service'
+
+import useAxios from '~/hooks/use-axios'
+import { defaultResponses } from '~/constants'
+
+const useUserInfo = ({ fetchOnMount = true, id, role } = {}) => {
```

Comment:
why do we need this hook? where can we use it besides the profile?

Reply:
As for now only on the Profile page, however, in the future I assume it may be useful. Maybe we will need this functionality on the new pages. So, I would like to leave it as it is.

Reply:
I don't think there will be many pages in the future where we will use it.